MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -231,6 +238,46 @@ public void setWebReturnToLabel(String webReturnToLabel) {
     }
 
     /**
+     * Enables Android Pay support in the {@link BuyClient}
+     * @param androidPayPublicKey The base64 encoded public key associated with Android Pay.
+     */
+    public void enableAndroidPay(String androidPayPublicKey) {
+        if (TextUtils.isEmpty(androidPayPublicKey)) {
+            throw new IllegalArgumentException("androidPayPublicKey cannot be empty");
+        }
+
+        byte[] digest;
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            digest = messageDigest.digest(androidPayPublicKey.getBytes("UTF-8"));
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+            // Do not enable Android Pay if the hash could not be computed
+            return;
+        }
+
+        // Enable Android Pay by setting the hash and key
+        this.androidPayPublicKeyHash = Base64.encodeToString(digest, Base64.DEFAULT);
+        this.androidPayPublicKey = androidPayPublicKey;
+    }
+
+    public boolean androidPayIsEnabled() {
+        return !TextUtils.isEmpty(androidPayPublicKey);
+    }
+
+    public void disableAndroidPay() {
```

Comment:
enableAndroidPay should be public, it is needed to set the public key.
disableAndroidPay should be public as well, and it can be used to disable Android Pay on the buyClient if an unrecoverable error was returned during Checkout.
400 indicates that the hash or public key was malformed.  Subsequent tries will hit the same error so android pay should be disabled in the buyClient.
410 indicates that the public key could not be found on the server, so same as above subsequent checkout calls will not work.